### PR TITLE
front: Create test e2e rolling stock editor

### LINKED ISSUE #4569: front: refacto E2E tests: rs editor
# Description

Writing end-to-end tests for the new module of rolling stock editor

# Goal

Increase stability of the module

# Acceptance criteria

- [ ] The basic workflows (creation, edition, deletion) of the rolling stock editor have been tested
 
# Notes:
**Create a new RS :**
Click on the button “Nouveau materiel roulant”
Insert all the general parameters 
Click the button confirm 
Click on “courbe effort Vitesse” 
Put values about speeds (on km/h) and traction force (on daN) on the table

_Points deleted , out of test scope_
~~Check that points are added on the curve. Check that they have the good value.~~
~~Check that the function copy/paste works for one line or for the value on a case~~
~~Check that we can delete one line and that the linked point is deleted from the curve~~ 

Click on confirm
Assert that the new rolling stock is created (we can find it by searching into the RS list). Assert that all the properties have the expected value.

**Change an existing RS :** 
Click on the button “duplicate” on a certain RS, assert the message appears (“sohuaitez-vous confirmer cette action?”, click on yes
Assert a new material is created with the same name + “copie”
Click on the button to edit 
Repeat all the actions above 
Should we test also the different curves for all the profiles??

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
- Add test for the creation of new rolling stock
 - Add test for duplation of rolling stock and update its name

close #4569

### CODE REVIEW

File: front/src/common/BootstrapSNCF/InputSNCF.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ export type InputSNCFProps = {
   min?: string | number;
   max?: string | number;
   list?: string;
+  dataTestID?: string;
```
Comment (Caracol3) [inline front/src/common/BootstrapSNCF/InputSNCF.tsx]: Can you add it in the input ?

**Review by Caracol3 — COMMENTED** [review]
I just started the review . I will continue it tomorrow

File: front/src/common/BootstrapSNCF/InputSNCF.tsx (hunk 1 above)
Comment (Uriel-Sautron) [reply front/src/common/BootstrapSNCF/InputSNCF.tsx]: Done

**Review by Caracol3 — APPROVED** [review]
lgtm !

File: front/tests/010-rollingstock-editor.spec.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('length', 50);
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('mass-input', 80, true);
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('maxSpeed', 200);
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('startupTime', 5);
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('startupAcceleration', 5);
+
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('comfortAcceleration', 1);
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('inertiaCoefficient', 1);
+    await playwrightRollingstockEditorPage.fillAndCheckInputById('gammaValue', 1);
```
Comment (SharglutDev) [inline front/tests/010-rollingstock-editor.spec.ts:35]: For all these repetitive lines, couldn't we loop on an array or object with all needed parameters ?
Like : 
```
const rollingStockTestForm = [["name", "Test_name"], ["detail", "Testdetail"]]
or
const rollingStockTestForm = [{
      property: "name", 
      label: "Test_name"
   }, {
      property: "detail", 
      label: "Test detail"
   }
]
```

File: front/tests/010-rollingstock-editor.spec.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+    await expect(powerRestrictionSelector.getByTitle('C1')).toBeVisible();
+
+    await playwrightRollingstockEditorPage.setSpreedsheetCell('0', velocityCellRow0);
+
+    await playwrightRollingstockEditorPage.setSpreedsheetCell('30500', effortCellRow0);
+
+    await playwrightRollingstockEditorPage.setSpreedsheetCell('5', velocityCellRow1);
+
+    await playwrightRollingstockEditorPage.setSpreedsheetCell('30000', effortCellRow1);
```
Comment (SharglutDev) [inline front/tests/010-rollingstock-editor.spec.ts:112]: Same here to easily fill a spreadsheet

**Review by SharglutDev — COMMENTED** [review]
Thank you for your PR ! Left some (debatable) suggestions

File: front/tests/010-rollingstock-editor.spec.ts (hunk 2 above)
Comment (Uriel-Sautron) [reply front/tests/010-rollingstock-editor.spec.ts:35]: As we've seen, async is not well managed in loops.

File: front/tests/010-rollingstock-editor.spec.ts (hunk 3 above)
Comment (Uriel-Sautron) [reply front/tests/010-rollingstock-editor.spec.ts:112]: Same here

File: front/src/common/Selector/Selector.tsx [hunk 4]
```diff
@@ -143,6 +143,7 @@ const Selector = <
       selectNewItem: (arg: NonNullable<T>) => void;
       customOnClick?: () => void;
     };
+    dataTestID?: string;
```
Comment (Math-R) [inline front/src/common/Selector/Selector.tsx]: dataTestId

File: front/src/common/BootstrapSNCF/InputGroupSNCF.tsx [hunk 5]
```diff
@@ -125,7 +127,7 @@ export default function InputGroupSNCF({
         value={inputValue}
         min={min}
         max={max}
-        data-testid="input-group-first-field"
+        data-testid={inputDataTestId}
```
Comment (nicolaswurtz) [inline front/src/common/BootstrapSNCF/InputGroupSNCF.tsx:130]: Why use camelCase here where you use kebab-case above?

File: front/src/common/Selector/Selector.tsx [hunk 6]
```diff
@@ -156,12 +157,13 @@ const Selector = <
...
     extraColumn,
     selectNewItemButtonProps,
+    dataTestID,
   } = props;
 
   const [hoveredItem, setHoveredItem] = useState<string | null>(null);
 
   return (
-    <div className="selector">
+    <div className="selector" data-testid={dataTestID}>
```
Comment (nicolaswurtz) [inline front/src/common/Selector/Selector.tsx]: Same here

File: front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormHelpers.tsx [hunk 7]
```diff
@@ -145,6 +145,7 @@ const RollingStockEditorParameterFormColumn = ({
           >
             <InputGroupSNCF
               id={property.title}
+              inputDataTestId={`${property.title}-input`}
```
Comment (nicolaswurtz) [inline front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormHelpers.tsx]: camelCase or kebab-case for property names?

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]
All seems to be good for me, except property & class names: you've to harmonize and choose one nomenclature for all names (testid / testId / testID for example). Thx!

File: front/src/common/Selector/Selector.tsx (hunk 4 above)
Comment (Uriel-Sautron) [reply front/src/common/Selector/Selector.tsx]: Done

File: front/src/common/BootstrapSNCF/InputGroupSNCF.tsx (hunk 5 above)
Comment (Uriel-Sautron) [reply front/src/common/BootstrapSNCF/InputGroupSNCF.tsx:130]: Here camelCase is for the props variable

File: front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormHelpers.tsx (hunk 7 above)
Comment (Uriel-Sautron) [reply front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormHelpers.tsx]: Here inputDataTestId is props not a property

**Review by Math-R — APPROVED** [review]

**Review by nicolaswurtz — APPROVED** [review]

**Pull request merged**